Themis/ViewModels/Assessment/Modeling/UMLRendererViewModel.swift

```diff
@@ -103,30 +106,6 @@ class UMLRendererViewModel: ExerciseRendererViewModel {
         undoManager.removeAllActions()
     }
     
-    @MainActor
-    func render(_ context: inout GraphicsContext, size: CGSize) {
-        guard let model = umlModel,
-              let modelType = model.type,
-              !diagramTypeUnsupported else {
-            return
-        }
-        let umlContext = UMLGraphicsContext(context)
-        let canvasBounds = CGRect(x: 0, y: 0, width: size.width, height: size.height)
-        
-        var renderer = UMLDiagramRendererFactory.renderer(for: modelType,
-                                                          context: umlContext,
-                                                          canvasBounds: canvasBounds,
-                                                          fontSize: fontSize)
-        
-        if let renderer {
-            renderer.render(umlModel: model)
-        } else {
-            log.error("Attempted to draw an unknown diagram type")
-            diagramTypeUnsupported = true
-            setError(.diagramNotSupported)
```

Comment:
For unsupported diagram types, we need to show an error message to the user. Since `diagramTypeUnsupported` is now calculated in `ApollonViewModel`, `UMLRendererViewModel` does not know if a diagram type can be supported. Consequently, the error can't be shown.

One way to solve this could be to have a static function in `ApollonViewModel` that takes a `UMLDiagramType` as a parameter and returns true if Apollon-iOS can render it. We could then use this function in `setup(basedOn:...)` to determine if the diagram type is supported.

Reply:
True! I added a static function to the `UMLDiagramType` enum, as it can be easily found and edited when more diagram support is available.